lib/version.js

```diff
@@ -11,18 +11,31 @@ const semver = require('semver');
 
 // Given a list of versions (as strings), return the latest version.
 // Return undefined if no version could be found.
-function latest(versions) {
+function latest(versions, pre = false) {
   let version = '';
   let origVersions = versions;
-  versions = versions.filter(function(version) {
-    return (/[0-9]/).test(version);
+  versions = origVersions.filter(function(version) {
+    return (/\d+\.\d+/).test(version);
   });
+  if (!versions.length){
+    versions = origVersions.filter(function(version) {
+      return (/\d+/).test(version);
+    });
+  }
+  if (!pre){
+    versions = versions.filter(function(version) {
+      return !(/\d+-\w+/).test(version);
+    });
```

Comment:
This is a bit difficult to follow.

I understand the purpose served by filtering out the prereleases. Could you explain what the rest of the filtering is for? Would it not work to just pass the other versions to the loose compare function?

Reply:
I have the `\d+\.\d+` test first to get what we would assume is definitely a version.
If that returns no results, we fall back to just look for anything that could possibly be a version.

Otherwise in an array of:
```json
["abc123f312", "1.2.3", "1.3"]
```
`abc123f312` would be the newest version as it's interpreted as `123`